src/core/payment-status.ts

```diff
@@ -0,0 +1,37 @@
+import { Result, ResultAsync } from "neverthrow"
+import { InvoiceUser } from "@services/mongoose/schema"
+import { decodeInvoice } from "@domain/ln-invoice"
+import { toTypedError } from "@domain/utils"
+
+export const MakePaymentStatusChecker = ({
+  paymentRequest,
+}): Result<PaymentStatusChecker, LnInvoiceDecodeError> => {
+  const decodedInvoice = decodeInvoice(paymentRequest)
+  const getStatus = (): ResultAsync<PaymentStatus, PaymentStatusError> => {
+    return decodedInvoice
+      .asyncAndThen(({ paymentHash }) => findUserInvoice(paymentHash))
+      .map((invoice) => ({
```

Comment:
Is there a need for an array here?

Reply:
This `.map` function is being called on the [`ResultAsync`](https://github.com/supermacro/neverthrow#resultasyncmap-method-1) type - its not an array.